src/canvas/canvas.service.ts

```diff
@@ -123,4 +157,31 @@ export class CanvasService {
       (this.produceEmptyCanvasState(filter.canvasId) as CanvasStateEntity) //If state is empty return default one
     );
   }
+
+  public async giveAccess(command: GiveAccessCommand) {
+    const user = await this.userRepository.findOne({
+      where: { id: command.userId },
+    });
+    const canvas = await this.canvasRepository.findOne({
+      where: { id: command.canvasId },
+    });
+    let canvasAccess = await this.canvasAccessRepository.findOne({
+      where: { user: { id: command.userId }, canvas: { id: command.canvasId } },
+    });
+    if (canvasAccess) {
```

Comment:
Wygląda na to że tutaj brakuje jeszcze sprawdzenia czy podany user i canvas istnieją. Jeżeli podam w body id canvasu jakiego nie ma w bazie to pewnie 500 poleci, bo będzie próbowała appka zapisać nulla w kolumnie not null